nixos/modules/services/web-apps/healthchecks.nix

```
@@ -0,0 +1,202 @@
+{ config, lib, pkgs, buildEnv, ... }:
+
+with lib;
+
+let
+  cfg = config.services.healthchecks;
+  pkg = cfg.package;
+  boolToPython = b: if b then "True" else "False";
+  environment = cfg.settings // {
+    PYTHONPATH = pkg.pythonPath;
+    STATIC_ROOT = cfg.dataDir + "/static";
+    DB_NAME = "${cfg.dataDir}/healthchecks.sqlite";
+    ALLOWED_HOSTS = lib.concatStringsSep "," cfg.settings.ALLOWED_HOSTS;
+    DEBUG = boolToPython cfg.settings.DEBUG;
+    REGISTRATION_OPEN = boolToPython cfg.settings.REGISTRATION_OPEN;
+  };
```

```suggestion
  environment = {
    PYTHONPATH = pkg.pythonPath;
    STATIC_ROOT = cfg.dataDir + "/static";
    DB_NAME = "${cfg.dataDir}/healthchecks.sqlite";
    ALLOWED_HOSTS = lib.concatStringsSep "," cfg.settings.ALLOWED_HOSTS;
    DEBUG = boolToPython cfg.settings.DEBUG;
    REGISTRATION_OPEN = boolToPython cfg.settings.REGISTRATION_OPEN;
  } // cfg.settings;
```

I am not sure if we want to people to overwrite those settings or not.

---

If you set those down in `config` then they can be overwritten with `mkForce` which is my personal preference.

---

I've applied the suggestion, cfg.settings overrides environment now.

The problem before was that i.e. settings.DEBUG (a boolean) would have overridden environment.DEBUG (a python string containing said boolean). 
I believe I solved that by using mkOptions `apply` for ALLOWED_HOSTS, DEBUG & REGISTRATION_OPEN.